R/FilterStateChoices.R

```diff
@@ -168,10 +168,11 @@ ChoicesFilterState <- R6::R6Class( # nolint
         x <- factor(as.character(x), levels = as.character(sort(unique(x))))
       }
       x <- droplevels(x)
-      choices <- table(x)
-      private$set_choices(names(choices))
-      self$set_selected(names(choices))
-      private$set_choices_counts(unname(choices))
+      choices_table <- table(x)
+      private$set_choices(names(choices_table))
+      self$set_selected(names(choices_table))
+      private$set_choices_counts(unname(choices_table))
+      private$set_choices_limited(x, choices)
```

Comment:
Looks like `choices` are not set anywhere. `private$choices` and `private$choices_count` are  set on values taken from `x`. As a result `FilterState` still have all choices but with the flag `choices_limited`.

What I would suggest is to filter `x` and `x_reactive` in the beginning constructor and set flag if choices are not complete. 

Also - since `x` is a vector, `choices` argument could have a default value `choices = unique(na.omit(x))` in this class.